File: RaWin/ViewModels/MonitorCategoryViewModel.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace RaWin.ViewModels
{
    /// <summary>
    /// ViewModel for a single monitoring card/category.
    /// </summary>
    public class MonitorCategoryViewModel : INotifyPropertyChanged
    {
        public string Name { get; set; }
        public StatusPanelViewModel StatusPanel { get; set; }
        public MetricsPanelViewModel MetricsPanel { get; set; }
        public LogsPanelViewModel LogsPanel { get; set; }
        public CommandsPanelViewModel CommandsPanel { get; set; }

        public bool IsStatusPanel { get; set; }
        public bool IsMetricsPanel { get; set; }
        public bool IsLogsPanel { get; set; }
        public bool IsCommandsPanel { get; set; }

        public event PropertyChangedEventHandler PropertyChanged;
        protected void OnPropertyChanged([CallerMemberName] string prop = null)
            => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(prop));
    }
}

File: RaStudios/viewmodels/monitor_category_viewmodel.py
from viewmodels.observable_object import ObservableObject

class MonitorCategoryViewModel(ObservableObject):
    """
    ViewModel for a single monitoring card/category.
    Supports plugin/extension category panels and agentic diagnostics.
    """
    def __init__(self, name=None):
        super().__init__()
        self.name = name
        self.status_panel = None
        self.metrics_panel = None
        self.logs_panel = None
        self.commands_panel = None
        self.is_status_panel = False
        self.is_metrics_panel = False
        self.is_logs_panel = False
        self.is_commands_panel = False
        # Add plugin/extension category support and diagnostics context as needed